### ISSUE TITLE
UI Improvement: ListView SwipeMenu

### ISSUE THREAD

**Issue by sonicnkt** [aminecmi/ReaderforSelfoss#301]
### Prerequisites

* [X] Are you running the latest version?
* [X] Did you check for an existing issue ?
* [X] Are you reporting to the correct repository?
* [X] Did you perform a cursory search?
* [X] Did you read the `CONTRIBUTING` guide ?

### Description

The Reader allready supports the swipe right gesture to mark items as read. I was thinking it would be cool to improve this even more.  My idea would be to add a swipe menu to the swipe left gesture which is currently not used (or make this configureable in the settings). https://androidexample365.com/a-swipe-menu-for-listview/

I would propose two menu items:
1. Save to favorites
2. Mark everything above as read

The second option should mark every item in the current view above the item as read. I think this would be great when you browsed through the list but have not viewed all listed items yet so you can carry on later from that point and dont have to mark every single item individually or browse down again to the point you left. 
I am also not sure if this is possible with the current selfoss api, as a feature like that is not available in the browser version but im planning to propose this there as well.

### Screenshots 

I have created a small mockup which show this menu integrated in the listview.

![sr_mockup_swipe-menu_1](https://user-images.githubusercontent.com/5454622/81153025-cd3bf300-8f82-11ea-8cf9-b0ade31049cf.png)
![sr_mockup_swipe-menu_2](https://user-images.githubusercontent.com/5454622/81153035-cf05b680-8f82-11ea-81fa-99de39bc7a78.png)

**Comment by aminecmi** [comment]
Hello @sonicnkt and sorry for the late reply,

There is currently a bar that is displayed if you press or long press the item (there is an option to choose the action).

Moreover, the "swipe menu" element you are proposing does not match any element of the material design specification.

If the action bar is not what you are looking fore, I could use something like [this](https://material.io/components/sheets-bottom#usage).

What do you think ?

**Comment by sonicnkt** [comment]
Hi no problem.
Pressing  / Longpressing an item does not show a bar on the latest fdroid build. Ateleast not on my devices (tried list and card view) so i dont know what this looks like.

I am not commited to the swipe menu if this breaks the design specifications i just saw this in another app and thought it would be a good idea. 
The bottom sheet method looks good as well. 

btw, is the mark everything above possible? other contents for the actiobar or bottom sheet should be mark as favorite and share?

**Comment by aminecmi** [comment]
Well, that's awkward. I may have removed the action bar.

I'll use the bottom sheet to replace it, then.

> btw, is the mark everything above possible? 

Yep, it should be possible, but maybe a little too complicated. Would something like "hide on scroll" would fit what you want ?

> other contents for the actiobar or bottom sheet should be mark as favorite and share?

The old one had: share, favorite, open in browser. Should I add something else ?

**Comment by sonicnkt** [comment]
ah ok, that explains this... 😄 

Those options should be enough!

hide on scroll would be lost after an app restart by the sound of it and not synced to the server. 
So not really a solution for my case but i can understand that its complicated to implement. I proposed this for selfoss server as well. If there is an official method to mark multiple items as read through the api it might be easier to implement this. 
Currently i guess you would have to collect every item from the current view (filter) with a newer timestamp as the one selected and then mark all of those as read individually in the background?

**Comment by aminecmi** [comment]
> hide on scroll would be lost after an app restart 
No it won't ! That's the point. Each time an element is out of the screen, I call the server to mark the item as read.

**Comment by sonicnkt** [comment]
ah ok i think i got it, so scrolling down would mark everything as read that is not visible anymore?
Would it be possible to create a kind of buffer for this that you could recover the last 5 items that are of screen for example if you scrolled a bit to fast?

**Comment by aminecmi** [comment]
What do you mean "recover" ? Like have some timeout that mark the items as read only after X seconds, so you could scroll back to them ?

**Comment by sonicnkt** [comment]
i thougt more about a kind of buffer that the only the 6. item that went ofscreen would be mark as read so that you could scroll up again 5 items. 

But a timeout would be even better if this is possible i guess.

**Comment by aminecmi** [comment]
Ok ! I'll look into any one of those solutions. I'll create a new issue for this since it's not an action for the list view anymore.

**Comment by sonicnkt** [comment]
Thanks you very much for looking into this :)
